picard/ui/itemviews.py

```diff
@@ -259,13 +264,15 @@ def update_selection(self, i, j):
     def update_selection_0(self):
         if not self._ignore_selection_changes:
             self._ignore_selection_changes = True
-            self.update_selection(0, 1)
+            if not self._processing:
+                self.update_selection(0, 1)
```

Comment:
What about disconnecting signals instead?

Something like:

```python
    def __init__(self, window, parent=None):
        super().__init__(parent)
        self.window = window
        self.create_icons()
        self.views = [FileTreeView(window, self), AlbumTreeView(window, self)]
        self._selected_view = 0
        self._ignore_selection_changes = False
        self.set_processing(False)
```

```python
    def set_processing(self, processing=True):
        self._processing = processing
        if processing:
            self.views[0].itemSelectionChanged.disconnect()
            self.views[1].itemSelectionChanged.disconnect()
        else:
            self.views[0].itemSelectionChanged.connect(self.update_selection_0)
            self.views[1].itemSelectionChanged.connect(self.update_selection_1)
```

```python
    def update_selection_0(self):
        if not self._ignore_selection_changes:
            self._ignore_selection_changes = True
            self.update_selection(0, 1)
            self._ignore_selection_changes = False

    def update_selection_1(self):
        if not self._ignore_selection_changes:
            self._ignore_selection_changes = True
            self.update_selection(1, 0)
            self._ignore_selection_changes = False
```

Reply:
Trying it out, but it does seems to be a better idea (unless the connect/disconnect costs are higher than checking the event and skipping the update_selection).

Reply:
Just realized that I'm dumb. It's being used the same way as self._ignore_selection_changes. Replacing self._processing references to _ignore_selection_changes.

Reply:
OK :)